source/grid/tria.cc

```diff
@@ -2192,15 +2227,10 @@ namespace internal
                                       line->vertex_index(1),
                                       subcell_line.boundary_id));
                       }
-                    else
-                      {
-                        line->set_manifold_id(subcell_line.manifold_id);
-                      }
```

Comment:
Now that this `else` statement has been redone the previous block doesn't really make sense: whether or not a manifold id is set has nothing to do with the boundary id of the line. Would you mind refactoring this check (i.e., lines 2202 through 2334)?